tests/lib/MB/EditNote.test.ts

```diff
@@ -0,0 +1,128 @@
+import { EditNote } from '@lib/MB/EditNote';
+
+let textarea: HTMLTextAreaElement;
+
+beforeEach(() => {
+    document.body.innerHTML = '<textarea class="edit-note" />';
+    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+    textarea = document.body.querySelector<HTMLTextAreaElement>('textarea')!;
+});
+
+describe('adding extra info', () => {
+    it('adds the info line', () => {
+        const editNote = new EditNote('footer');
+        editNote.addExtraInfo('info line');
+
+        expect(textarea.value.trim()).toBe('info line');
+    });
+
+    it('retains previous content', () => {
+        const editNote = new EditNote('footer');
+        textarea.value = 'preamble';
+        editNote.addExtraInfo('info line');
+
+        expect(textarea.value).toBe('preamble\ninfo line');
+    });
+
+    it('filters out duplicate lines', () => {
+        const editNote = new EditNote('footer');
+        editNote.addExtraInfo('info line');
+        editNote.addExtraInfo('info line');
+
+        expect(textarea.value.trim()).toBe('info line');
+    });
+
+    it('retains different lines', () => {
+        const editNote = new EditNote('footer');
+        editNote.addExtraInfo('info line');
+        editNote.addExtraInfo('info line 2');
+
+        expect(textarea.value.trim()).toBe('info line\ninfo line 2');
+    });
+
+    it('filters out pre-existing duplicates', () => {
+        textarea.value = 'info line';
+        const editNote = new EditNote('footer');
+        editNote.addExtraInfo('info line');
+
+        expect(textarea.value.trim()).toBe('info line');
+    });
+
+    it('retains pre-existing lines', () => {
+        textarea.value = 'preamble';
+        const editNote = new EditNote('footer');
+        editNote.addExtraInfo('info line');
+
+        expect(textarea.value.trim()).toBe('preamble\ninfo line');
+    });
```

Comment:
This looks almost identical to 'retains previous content', the only differences seem to be the order of the first two statements and whether `trim()` is called or not (which should have no effect here).

Reply:
They test different things. One tests whether it retains content that was already in the edit note before the instance was created, the other tests whether it retains content added to the edit not after the instance was created. I have to admit that it took me a couple of minutes to figure out the difference too, so I reordered the cases to put these closer together and renamed them to make the difference more noticeable.